hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/DirectoryEntityHandler.java

```diff
@@ -55,15 +58,34 @@ public NamespaceSummaryResponse getSummaryResponse()
           throws IOException {
     // path should exist so we don't need any extra verification/null check
     long dirObjectId = getBucketHandler().getDirObjectId(getNames());
+    CountStats countStats = new CountStats(
+        -1, -1,
+        getTotalDirCount(dirObjectId), getTotalKeyCount(dirObjectId));
     NamespaceSummaryResponse namespaceSummaryResponse =
-            new NamespaceSummaryResponse(EntityType.DIRECTORY);
-    namespaceSummaryResponse
-        .setNumTotalDir(getTotalDirCount(dirObjectId));
-    namespaceSummaryResponse.setNumTotalKey(getTotalKeyCount(dirObjectId));
+        NamespaceSummaryResponse.newBuilder()
+            .setEntityType(EntityType.DIRECTORY)
+            .setCountStats(countStats)
+            .setObjectDBInfo(getDirectoryObjDbInfo(getNames()))
+            .setStatus(ResponseStatus.OK)
+            .build();
 
     return namespaceSummaryResponse;
   }
 
+  private ObjectDBInfo getDirectoryObjDbInfo(String[] names)
+      throws IOException {
+    OmDirectoryInfo dirInfo = getBucketHandler().getDirInfo(names);
+    ObjectDBInfo objectDBInfo = new ObjectDBInfo();
```

Comment:
Same as `BucketObjectDBInfo`, create constructor or builder which takes `OmDirectoryInfo` and create `ObjectDBInfo` object.

Reply:
Thanks @hemantk-12 for review. Changes done. Pls re-review.